server/feedback/src/core.rs

```diff
@@ -0,0 +1,53 @@
+use crate::tokens::Claims;
+use actix_web::web::Data;
+use actix_web::HttpResponse;
+use jsonwebtoken::{encode, EncodingKey, Header};
+use log::error;
+
+use tokio::sync::Mutex;
+
+pub struct AppStateFeedback {
+    pub(crate) feedback_keys: crate::FeedbackKeys,
+    pub(crate) token_record: Mutex<Vec<TokenRecord>>,
+}
+impl AppStateFeedback {
+    pub fn from(feedback_keys: crate::FeedbackKeys) -> AppStateFeedback {
+        AppStateFeedback {
+            feedback_keys,
+            token_record: Mutex::new(Vec::new()),
+        }
+    }
+    pub fn able_to_process_feedback(&self) -> bool {
+        self.feedback_keys.github_token.is_some() && self.feedback_keys.jwt_key.is_some()
+    }
+}
+
+pub struct TokenRecord {
+    pub kid: u64,
+    pub next_reset: usize,
+}
+
+pub async fn get_token(state: Data<AppStateFeedback>) -> HttpResponse {
```

Comment:
Would it make sense to move `get_token` into `tokens.rs`?

Reply:
True.
I have no idea why I came up with this architecture in the first place.
I have refactored this to live in the token directory and simplified the acompaniing code